src/sentinel.c

```diff
@@ -1993,8 +2023,13 @@ void sentinelSendAuthIfNeeded(sentinelRedisInstance *ri, redisAsyncContext *c) {
         auth_pass = ri->master->auth_pass;
         auth_user = ri->master->auth_user;
     } else if (ri->flags & SRI_SENTINEL) {
-        auth_pass = server.requirepass;
-        auth_user = NULL;
+        if (sentinel.sentinel_auth_user && sentinel.sentinel_auth_pass) {
+            auth_pass = sentinel.sentinel_auth_pass;
+            auth_user = sentinel.sentinel_auth_user;
+        } else {
+            auth_pass = server.requirepass;
+            auth_user = NULL;
+        }
```

Comment:
so `requirepass` is only compatibility with old config files, and should no longer be used once ACL is used, right?
if that's the case, let's document it in a few places.

also, maybe if the user just configured `sentinel_auth_pass` and din't configure `sentinel_auth_user` maybe we should use "default" user? (i.e. the above `if` should just look for the `pass`, not `user &&`.

Reply:
@oranagra right, I do find few places mentioning requirepass is used for auth between sentinels in sentinel.conf. Let me fix these. For the second part, I am not sure, if we only check pass, and if user is empty, that means we use default user with sentinel.sentinel_auth_pass to auth. I felt like this may make the interface more complicated. Maybe force the ACL auth way to be valid only both sentinel.sentinel_auth_pass and sentinel.sentinel_auth_user are set?

Reply:
i think it's kinda like `masterauth` and `masteruser`.
if `masteruser` is missing, then it uses an AUTH with single argument.

the difference is that in redis, there was a clear distinction between `requirepass` and `masterauth`, but in sentinel `requirepass` used to do both (both require a password for incoming connections, and also using the same password to connect to others).

but if we ignore that difference, and consider the sentinel's `requirepass` being similar to redis's `masterauth`, then with the absence of a user name, it can use an AUTH without one.

Reply:
@hwware what did we decide about this one? do we wanna make it similar to masterauth? (look at my last sentence above)

Reply:
thanks @oranagra , please review my new changes for this part, thanks!